controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -152,125 +164,287 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
+		return err
+	}
+
+	// and finally cast all these configs and create the secret
+	cdEnv := make(map[string][]byte)
+	cdEnv["cdenvconfig.json"] = envData
+	iqeSecret.Data = cdEnv
+	if err := cache.Update(CoreIqeSecret, iqeSecret); err != nil {
+		r.Log.Error(err, "Failed to check for iqe secret", "clowdjobinvoaction")
+		return err
+	}
+
+	return nil
+}
+
+func (r *ClowdJobInvocationReconciler) createIqeJobResource(cache *providers.ObjectCache, cji *crd.ClowdJobInvocation, env *crd.ClowdEnvironment, app *crd.ClowdApp, nn types.NamespacedName, ctx context.Context, j *batchv1.Job) error {
 	labels := cji.GetLabels()
 	cji.SetObjectMeta(j, crd.Name(nn.Name), crd.Labels(labels))
 
 	j.ObjectMeta.Labels = labels
 	j.Spec.Template.ObjectMeta.Labels = labels
 
-	pod := job.PodSpec
-
-	if job.RestartPolicy == "" {
-		j.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
-	} else {
-		j.Spec.Template.Spec.RestartPolicy = job.RestartPolicy
-	}
+	j.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
 
 	j.Spec.Template.Spec.ImagePullSecrets = []core.LocalObjectReference{
 		{Name: "quay-cloudservices-pull"},
 	}
 
-	envvar := pod.Env
-	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
+	pod := crd.PodSpec{
+		Resources: env.Spec.Providers.Testing.Iqe.Resources,
+	}
 
-	var livenessProbe core.Probe
-	var readinessProbe core.Probe
+	envvar := []core.EnvVar{}
+	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
+	envvar = append(envvar, core.EnvVar{Name: "ENV_FOR_DYNACONF", Value: cji.Spec.Testing.Iqe.DynaconfEnvName})
+	envvar = append(envvar, core.EnvVar{Name: "NAMESPACE", Value: nn.Namespace})
+	envvar = append(envvar, core.EnvVar{Name: "CLOWDER_ENABLED", Value: "true"})
 
-	if pod.LivenessProbe != nil {
-		livenessProbe = *pod.LivenessProbe
+	tag := ""
+	if cji.Spec.Testing.Iqe.ImageTag != "" {
+		tag = cji.Spec.Testing.Iqe.ImageTag
 	} else {
-		livenessProbe = core.Probe{}
+		tag = app.Spec.Testing.IqePlugin
 	}
-	if pod.ReadinessProbe != nil {
-		readinessProbe = *pod.ReadinessProbe
-	} else {
-		readinessProbe = core.Probe{}
+	plugin := app.Spec.Testing.IqePlugin
+	iqeImage := env.Spec.Providers.Testing.Iqe.ImageBase
+
+	accessLevel := env.Spec.Providers.Testing.K8SAccessLevel
+
+	switch accessLevel {
+	// Use edit level service account to create and delete resources
+	// one per app when the app is created
+	case "edit":
+		labeler := utils.GetCustomLabeler(nil, nn, app)
+		if err := svcAccounts.CreateServiceAccount(cache, svcAccounts.CoreAppServiceAccount, env.Spec.Providers.PullSecrets, nn, labeler); err != nil {
+			r.Recorder.Eventf(cji, "Warning", "ServiceAccountNotCreated", "Unable to create service account [%s]", nn.Name)
+			return err
+		}
+		j.Spec.Template.Spec.ServiceAccountName = nn.Name
+	// Standard view access to the owned resources
+	case "view":
+		j.Spec.Template.Spec.ServiceAccountName = app.Name
```

Comment:
I think this will have `-app` appended to it

Reply:
This also assumes that the app has "view" permissions - which by default it doesn't.......I think you need to create an iqe secret in this case also

Reply:
And thinking about it - The way that we are going to move for doing pull secrets - I think you'll ALWAYS need a service account - which will be populated with the pull secrets